source/fab/tasks/c.py

```diff
@@ -83,6 +88,12 @@ def _check_for_include(self, lineno) -> Optional[str]:
             return None
 
     def run(self, fpath: Path):
+
+        if not clang:
+            msg = 'clang not available, C analysis disabled'
+            warnings.warn(msg, ImportWarning)
+            return ImportWarning(msg)
```

Comment:
Note to reviewer: We use a multiprocessing pattern where child processes return either a successfully created object, or an error. Errors are then gathered and reported together. There's also a ticket on the [development board](https://github.com/orgs/metomi/projects/1/views/1) to review exception handling.